### PR TITLE
stream: fix idle state race condition (#1916)

### PR DESCRIPTION
**Pull request by leszko**
I think we should resurrect this PR from @gioelecerati . I'm testing it in staging now and it seems to work fine. I didn't experience any issues with webhooks or anything. That makes me think that maybe the webhook issues experienced by the customer were unrelated.

### CODE REVIEW

**Review by gioelecerati — APPROVED** [review]
Now with tests! LGTM

File: packages/www/components/StreamDetails/MultistreamTargetsTable/SaveTargetDialog.tsx [hunk 1]
```diff
@@ -233,7 +233,7 @@ const SaveTargetDialog = ({
   return (
     <AlertDialog open={isOpen} onOpenChange={onOpenChange}>
       <AlertDialogContent
-        css={{ maxWidth: 450, px: "$5", pt: "$4", pb: "$4" }}
+        css={{ maxWidth: 450, px: "$5", pt: "$4", pb: "$4", zIndex: 1 }}
```
Comment (gioelecerati) [inline packages/www/components/StreamDetails/MultistreamTargetsTable/SaveTargetDialog.tsx]: This is not needed, it was an unrelated change that got merged in another PR from @0xcadams

File: packages/api/src/controllers/stream.ts [hunk 2]
```diff
@@ -896,18 +896,17 @@ app.post(
...
+        : null;
     const patch: Partial<DBSession & DBStream> = {
-      isHealthy: payload.is_active ? payload.is_healthy : undefined,
+      isHealthy: payload.is_active ? payload.is_healthy : null,
       issues,
       // do not clear the `lastSeen` field when the stream is not active
       ...(payload.is_active ? { lastSeen: Date.now() } : null),
     };
 
     // Since we might need to delete some fields, use replace instead of update
```
Comment (victorges) [inline packages/api/src/controllers/stream.ts]: I think this comment can be removed

**Review by victorges — APPROVED** [review]
LGTM!

**Review by adamsoffer — APPROVED** [review]
LGTM

File: packages/www/components/StreamDetails/MultistreamTargetsTable/SaveTargetDialog.tsx (hunk 1 above)
Comment (leszko) [reply packages/www/components/StreamDetails/MultistreamTargetsTable/SaveTargetDialog.tsx]: Reverted in 614e408f7c7a468b4d321b6076a3d9ad1edfde45

File: packages/api/src/controllers/stream.ts (hunk 2 above)
Comment (leszko) [reply packages/api/src/controllers/stream.ts]: Removed in bd9959fa85ca4c1f86d1a638960c4e7e226cc161
